sympy/functions/special/delta_functions.py

```diff
@@ -31,6 +31,20 @@ class DiracDelta(Function):
 
     5) ``DiracDelta(x,k) = 0``, for all ``x != 0``
 
+    Examples
```

Comment:
Perhaps we can offload to Wikipedia for a formal definition of distributional functions. Here's the way I think of it for SymPy (feel free to reuse this text):

DiracDelta only makes sense in definite integrals, and in particular, integrals of the form `Integral(f(x)*DiracDelta(x - x0), (x, a, b))`, where it equals f(x0) if a <= x0 <= b and 0 otherwise. Formally, DiracDelta acts in some ways like a function that takes on oo at 0 and 0 elsewhere, but in many ways it also does not. It can often be useful to treat DiracDelta in formal ways, building up and manipulating expressions with delta functions (which may eventually be integrated), but care must be taken to not treat it as a real function. SymPy's `oo` is similar. It only truly makes sense formally in certain contexts (such as integration limits), but SymPy allows its use everywhere, and tries to consistent with operations on it (like 1/oo), but it is easy to get into trouble and get wrong results if `oo` is treated too much like a number. Similarly, if DiracDelta is treated too much like a function, it is easy to get wrong or nonsensical results.